src/components/LinkableText.tsx

```diff
@@ -9,27 +9,35 @@ interface Props {
 
 export const LinkableText: React.FC<Props> = ({text, ...props}) => {
   const handlePress = (url: string) => {
-    Linking.openURL(url);
+    Linking.canOpenURL(url).then((supported) => {
+      console.log(supported);
+      const regex = /(http|https)/;
+      if (!regex.test(url)) {
+        const urls = `http://${url}`;
+        Linking.openURL(urls);
+      }
+      Linking.openURL(url);
+    });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

There's a logical error in the `handlePress` function. If the URL does not start with 'http' or 'https', you add 'http://' to the beginning and open it, but then you also try to open the original URL. This will result in trying to open two URLs, one of which may not be valid. You should use an else statement to ensure only one URL is opened.
```diff
-      if (!regex.test(url)) {
-        const urls = `http://${url}`;
-        Linking.openURL(urls);
-      }
-      Linking.openURL(url);
+      if (regex.test(url)) {
+        Linking.openURL(url);
+      } else {
+        const urls = `http://${url}`;
+        Linking.openURL(urls);
+      }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->